library/ssl_msg.c

```diff
@@ -5505,6 +5519,37 @@ int mbedtls_ssl_write( mbedtls_ssl_context *ssl, const unsigned char *buf, size_
         }
     }
 
+    /* Dispatch any pending outgoing data to the underlying transport. */
+    if( ( ret = mbedtls_ssl_flush_output( ssl ) ) != 0 )
+    {
+        MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_flush_output", ret );
+        return( ret );
+    }
+
+    /* Truncate outgoing data to ensure that it fits in a record. */
+    ret = mbedtls_ssl_get_max_out_record_payload( ssl );
+    if( ret < 0 )
+    {
+        MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_get_max_out_record_payload", ret );
+        return( ret );
+    }
+    max_len = (size_t) ret;
+
+    if( len > max_len )
+    {
+#if defined(MBEDTLS_SSL_PROTO_DTLS)
+        if( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM )
+        {
+            MBEDTLS_SSL_DEBUG_MSG( 1, ( "fragment larger than the (negotiated) "
+                                        "maximum fragment length: %zu > %zu",
```

Comment:
To fix the CI issues, caused by `"%z"` being unsupported on some platforms:
```suggestion
                                        "maximum fragment length: %" MBEDTLS_PRINTF_SIZET
                                        " > %" MBEDTLS_PRINTF_SIZET,
```